algorithm.go

```diff
@@ -67,8 +68,10 @@ type extAlgorithm struct {
 	HashFunc func() hash.Hash
 }
 
-// extAlgorithms contains extended algorithms.
-var extAlgorithms map[Algorithm]extAlgorithm
+var (
+	extAlgorithms map[Algorithm]extAlgorithm
+	extMu         sync.RWMutex
+)
```

Comment:
@qmuntal I think [sync.Map](https://pkg.go.dev/sync#Map) is a better option than a map with a mutex.

Reply:
can we open another issue to account for this proposal?

Reply:
> @qmuntal I think [sync.Map](https://pkg.go.dev/sync#Map) is a better option than a map with a mutex.

I tend to avoid using `sync.Map` due to this comment in its docs:

> The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

But I agree that a `sync.Map` could also be used here and I wouldn't complain.

Reply:
Currently, there is no other invariants along with the map content. Well, it is a trade-off: readability and performance.

Using a plain map with a RWMutex is faster than sync.Map. Let's keep this PR unchanged.